salt/utils/gitfs.py

```diff
@@ -838,6 +895,20 @@ def _lock(self, lock_type="update", failhard=False):
         """
         Place a lock file if (and only if) it does not already exist.
         """
+        if self.__class__._master_lock.acquire(timeout=60) is False:
+            # if gitfs works right we should never see this timeout error.
+            log.error("gitfs master lock timeout!")
+            raise TimeoutError("gitfs master lock timeout!")
+        try:
+            return self.__lock(lock_type, failhard)
+        finally:
+            self.__class__._master_lock.release()
+
+    def __lock(self, lock_type="update", failhard=False):
+        """
+        Place a lock file if (and only if) it does not already exist.
+        Without MultiProcessing locks.
+        """
         try:
             fh_ = os.open(
```

Comment:
I'm a bit confused why `salt.utils.files.flopen` was not used there to prevent any possible race conditions with the file creation.

Reply:
Thanks for the review! And good point.

Hmmm, I have also no idea. Maybe this is something to have in mind and check in case we still detect more race conditions around this area.

For now, in the context of the present submission and CVE fix, I would say to proceed with the current changes. Later we can evaluate and test if we should also change that.